kodi_addon_checker/schema_validation.py

```diff
@@ -0,0 +1,115 @@
+    """
+
+    failed, metadatacount, validcount = _validation_checks(report, parsed_xml, branch_name)
+
+    root_schema = os.path.join(XML_SCHEMA, 'addon.xsd')
+    if not _validate(parsed_xml, root_schema):
+        report.add(Record(PROBLEM, "Schema validation failed for root addon element"))
+        validcount -= 1
+
+    if metadatacount == 0:
+        report.add(Record(PROBLEM, 'Metadata missing/occurred more than once'))
+
+    if validcount == 1:
+        report.add(Record(INFORMATION, "Valid XML file found"))
+    elif failed:
+        report.add(Record(PROBLEM, "Schema validation failed for the following points: %s " % ''.join(failed)))
+
+
+def _validation_checks(report: Report, parsed_xml, branch_name):
+    """Perform certain checks on the XML elements and extension point
+    """
+    metadatacount = 0
+    validcount = 1
+    failed = []
+    valid_points = {
+        'kodi.context.item': 'contextitem.xsd',
+        'kodi.game.controller': 'controller.xsd',
+        'kodi.resource.games': 'games.xsd',
+        'kodi.resource.images': 'images.xsd',
+        'kodi.resource.language': 'language.xsd',
+        'kodi.addon.metadata': 'metadata.xsd',
+        'kodi.resource.uisounds': 'uisounds.xsd',
+        'xbmc.addon.metadata': 'metadata.xsd',
+        'xbmc.addon.repository': 'repository.xsd',
+        'xbmc.gui.skin': 'skin.xsd',
+        'xbmc.metadata.scraper.albums': 'scraper.xsd',
+        'xbmc.metadata.scraper.artists': 'scraper.xsd',
+        'xbmc.metadata.scraper.movies': 'scraper.xsd',
+        'xbmc.metadata.scraper.musicvideos': 'scraper.xsd',
+        'xbmc.metadata.scraper.tvshows': 'scraper.xsd',
+        'xbmc.metadata.scraper.library': 'scraper.xsd',
+        'xbmc.python.script': 'script.xsd',
+        'xbmc.python.lyrics': 'script.xsd',
+        'xbmc.python.weather': 'script.xsd',
+        'xbmc.python.library': 'script.xsd',
+        'xbmc.python.pluginsource': 'pluginsource.xsd',
+        'xbmc.python.module': 'script.xsd',
+        'xbmc.service': 'service.xsd',
+        'xbmc.subtitle.module': 'script.xsd',
+        'xbmc.ui.screensaver': 'script.xsd',
+        'xbmc.webinterface': 'webinterface.xsd',
+    }
+
+    for extension in parsed_xml.findall("extension"):
+        extension_point = extension.get("point")
+
+        if extension_point in valid_points:
+            schema_path = check_version(branch_name, valid_points[extension_point])
+
+            if schema_path:
+                if extension_point == "xbmc.addon.metadata" or extension_point == "kodi.addon.metadata":
+                    metadatacount += 1
+
+                if not os.path.exists(schema_path):
+                    report.add(Record(PROBLEM, "%s doesn't exist" % schema_path))
+                    continue
+
+                if not _validate(extension, schema_path):
+                    failed.append(extension_point)
+                    validcount -= 1
+                parsed_xml.remove(extension)
```

Comment:
This changes `parsed_xml` passed at check_addon.py and causes an incomplete view of the addon.xml for later checks.

Reply:
While testing #108 onto this PR I wondered why no assets are available for the transparency checks. Following change proves my observation.
```
--- a/kodi_addon_checker/check_addon.py
+++ b/kodi_addon_checker/check_addon.py
@@ -43,5 +43,7 @@
             file_index = handle_files.create_file_index(addon_path)

+            print(ET.tostring(parsed_xml))
             schema_validation.schemas(addon_report, parsed_xml, branch_name)
+            print(ET.tostring(parsed_xml))

             check_dependencies.check_addon_dependencies(addon_report, repo_addons, parsed_xml, branch_name)
```